core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/hashJoin/HashJoinOpDesc.scala

```diff
@@ -22,17 +22,31 @@ import scala.collection.convert.ImplicitConversions.`collection AsScalaIterable`
 class HashJoinOpDesc[K] extends OperatorDescriptor {
 
   @JsonProperty(required = true)
-  @JsonSchemaTitle("Small Input attr")
-  @JsonPropertyDescription("Small Input Join Key")
+  @JsonSchemaTitle("Left Input Attribute")
+  @JsonPropertyDescription("attribute to be joined on the Left Input")
   @AutofillAttributeName
   var buildAttributeName: String = _
 
   @JsonProperty(required = true)
-  @JsonSchemaTitle("Large input attr")
-  @JsonPropertyDescription("Large Input Join Key")
+  @JsonSchemaTitle("Right Input Attribute")
+  @JsonPropertyDescription("attribute to be joined on the Right Input")
   @AutofillAttributeNameOnPort1
   var probeAttributeName: String = _
 
+  @JsonProperty(required = true, defaultValue = "false")
+  @JsonSchemaTitle("Left Outer Join?")
+  @JsonPropertyDescription(
+    "all records from the Left input, including the records that joined with the Right input"
+  )
+  var leftOuterJoin: Boolean = false
+
+  @JsonProperty(required = true, defaultValue = "false")
+  @JsonSchemaTitle("Right Outer Join?")
+  @JsonPropertyDescription(
```

Comment:
can we make this a drop down? inner / left outer / right outer / full other, use  a java enum, maybe called `JoinType`

Reply:
done.